test/gurka/test_recost.cc

```diff
@@ -49,19 +50,147 @@ TEST(recosting, same_historical) {
+                    R"("costing_options":{"auto":{"speed_types":[]}}})",
+                {}, &api);
+
+    // check we have the same cost at all places
+    for (const auto& n : api.trip().routes(0).legs(0).node()) {
+      EXPECT_EQ(n.recosts_size(), 1);
+      EXPECT_EQ(n.cost().elapsed_cost().seconds(), n.recosts(0).elapsed_cost().seconds());
+      EXPECT_EQ(n.cost().elapsed_cost().cost(), n.recosts(0).elapsed_cost().cost());
+      EXPECT_EQ(n.cost().transition_cost().seconds(), n.recosts(0).transition_cost().seconds());
+      EXPECT_EQ(n.cost().transition_cost().cost(), n.recosts(0).transition_cost().cost());
+      if (n.has_edge()) {
+        EXPECT_NEAR(n.edge().speed(), 55.f, .1f);
+      }
+    }
+  }
+
+  // set the time should use time dep forward and freeflow at 1am
+  {
+    Api api;
+    actor.route(R"({"costing":"auto","locations":[)" + locations +
+                    R"(],"recostings":[{"costing":"auto","name":"same"}],)" +
+                    R"("date_time":{"type":1,"value":"2020-08-01T01:23"}})",
+                {}, &api);
+
+    // check we have the same cost at all places
+    for (const auto& n : api.trip().routes(0).legs(0).node()) {
+      EXPECT_EQ(n.recosts_size(), 1);
+      EXPECT_EQ(n.cost().elapsed_cost().seconds(), n.recosts(0).elapsed_cost().seconds());
+      EXPECT_EQ(n.cost().elapsed_cost().cost(), n.recosts(0).elapsed_cost().cost());
+      EXPECT_EQ(n.cost().transition_cost().seconds(), n.recosts(0).transition_cost().seconds());
+      EXPECT_EQ(n.cost().transition_cost().cost(), n.recosts(0).transition_cost().cost());
+      if (n.has_edge()) {
+        EXPECT_NEAR(n.edge().speed(), 80.f, .1f);
+      }
+    }
+  }
+
+  // trivial route should use astar and constrained
+  {
+    std::string trivial = R"({"lon":)" + std::to_string(map.nodes["D"].lng()) + R"(,"lat":)" +
+                          std::to_string(map.nodes["D"].lat()) + R"(},{"lon":)" +
+                          std::to_string(map.nodes["B"].lng()) + R"(,"lat":)" +
+                          std::to_string(map.nodes["B"].lat()) + "}";
+
+    Api api;
+    actor.route(R"({"costing":"auto","locations":[)" + trivial +
+                    R"(],"recostings":[{"costing":"auto","name":"same"}]})",
+                {}, &api);
+
+    // check we have the same cost at all places
+    for (const auto& n : api.trip().routes(0).legs(0).node()) {
+      EXPECT_EQ(n.recosts_size(), 1);
+      EXPECT_EQ(n.cost().elapsed_cost().seconds(), n.recosts(0).elapsed_cost().seconds());
+      EXPECT_EQ(n.cost().elapsed_cost().cost(), n.recosts(0).elapsed_cost().cost());
+      EXPECT_EQ(n.cost().transition_cost().seconds(), n.recosts(0).transition_cost().seconds());
+      EXPECT_EQ(n.cost().transition_cost().cost(), n.recosts(0).transition_cost().cost());
+      if (n.has_edge()) {
+        EXPECT_NEAR(n.edge().speed(), 10.f, .1f);
+      }
+    }
+  }
+
+  // TODO: There's a difference in the way a route is costed in reverse vs forward this could be
```

Comment:
Is there an issue in the tracker for this?

Reply:
no i only just discovered it. i will add an issue to github